WordPress/Classes/Stores/NoticeStore.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import WordPressFlux
+
+struct Notice {
+    let title: String
+    let message: String?
+    let actionTitle: String?
+    let actionHandler: (() -> Void)?
+
+    init(title: String, message: String? = nil, actionTitle: String? = nil, actionHandler: (() -> Void)? = nil) {
+        self.title = title
+        self.message = message
+        self.actionTitle = actionTitle
+        self.actionHandler = actionHandler
+    }
+}
+
+
+enum NoticeAction: Action {
+    case post(Notice)
+    case dismiss(Notice)
+}
+
+
+struct NoticeStoreState {
```

Comment:
I've been thinking about some things that I find odd in the code, and I think it's just that the store is inspecting the values on the queue instead of just using push/pop. I think it might improve a bit if the store state was just the current notice to be presented (if any), and keep the rest in a queue of pending notices.

```swift
struct NoticeStoreState {
    var notice: Notice?
}


class NoticeStore: StatefulStore<NoticeStoreState> {
    private var pending = Queue<Notice>()

    // ...

    var nextNotice: Notice? {
        return state.notice
    }

    private func enqueueNotice(_ notice: Notice) {
        if state.notice == nil {
            state.notice = notice
        } else {
            pending.push(notice)
        }
    }

    private func dequeueNotice() {
        state.notice = pending.pop()
    }
}
```

This means the store would only dispatch state changes when the notice to present changes, and not every time a new one is enqueued, but honestly what threw me off was the `@discardableResult` on `Queue.pop()` 😁

Reply:
Ah, I like this! That makes sense, and you're right it means we only notify on changes for new notifications, not dismissing.